frontend/src/app/components/ExperimentPage.js

```diff
@@ -48,14 +48,15 @@ export default class ExperimentPage extends React.Component {
 
   componentWillReceiveProps(nextProps) {
     const { shouldShowTourDialog, enabled: prevEnabled } = this.state;
-    const { isExperimentEnabled, getExperimentBySlug } = nextProps;
 
-    const prevExperiment = getExperimentBySlug(this.props.params.slug);
+    const prevExperiment = this.props.getExperimentBySlug(this.props.params.slug);
     const prevInProgress = prevExperiment && prevExperiment.inProgress;
 
-    const nextExperiment = getExperimentBySlug(nextProps.params.slug);
+    const { isExperimentEnabled: nextIsExperimentEnabled,
+            getExperimentBySlug: nextGetExperimentBySlug } = nextProps;
```

Comment:
Oh, but I think I will back out this destructuring magic to be more verbose with the `nextProps` reference. Probably trying to be too clever, here